doc_internal/rust-devel.md

```diff
@@ -144,9 +146,15 @@ These types should be confined to the FFI modules, in particular `wchar_ffi`. Th
 
 ### Format strings
 
-Rust's builtin `std::fmt` modules do not accept runtime-provided format strings, so we mostly won't use them, except perhaps for FLOG / other non-translated text.
+There are two format string syntaxes:
+
+1. The standard `std::fmt` syntax. The `wformat!` macro creates a `WString`, in the same manner that `format!` constructs a a `String`.
+2. An `sprintf!` implementation for translated content, using printf-style format strings. This is necessary because Rust's builtin `std::fmt` modules do not accept runtime-provided format strings.
+
+Prefer to use `wformat!` (or `write!` to append to an existing `WString`) for content which will never be translated, and `sprintf!` or `wgettext_fmt!` for content that is translated or may be in the future.
 
-Instead we'll continue to use printf-style strings, with a Rust printf implementation.
+> **Warning**
+`WString` does not implement the width, fill, and alignment features of Rust formatting. Use the `to_string()` function to convert a `WString` to a `String` to apply a width, fill or alignment. `sprintf!` format strings do not have this limitation.
 
 ### Vectors
 
```

Comment:
This is unfortunate, especially because it silently drops width for WString format args.
I think this is acceptable for now but it would be nice if we can come up with a way to panic instead, or a clippy lint